### PR TITLE
MAINT: update sphinx to include sphinx v4

### PR DESCRIPTION
**Pull request by mmcky**
@AakashGfude I am trying to upgrade this package to be compatible with `sphinx4` but it has a testing dependency on `jupyter-book`. Having an upstream dependency makes these sort of updates difficult. Can we link up sometime to discuss having this dependency?

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -31,13 +31,13 @@ include_package_data = True
...
 myst =
-    myst-nb>=0.11,<0.13
+    myst-nb>=0.12,<0.14
 rtd =
     myst-parser
     sphinx-book-theme
 testing =
     coverage<5.0
-    jupyter-book>=0.11.0a1,<0.12
+    jupyter-book>=0.11,<0.12
```
Comment (mmcky) [inline setup.cfg:40]: @AakashGfude this is an issue as `jupyter-book` wont be updated to include `sphinx4` compatibility until this package is checked and updated :-)

Comment (mmcky) [reply setup.cfg:40]:
```
▶ grep -r "jupyter-book" .       
./sphinx_jupyterbook_latex.egg-info/requires.txt:jupyter-book<0.12,>=0.11.0a1
./CHANGELOG.md:1. Make this package a "standalone" sphinx extension, independent of jupyter-book, i.e. not dependant on the presence of a `_toc.yml`.
./CHANGELOG.md:This is the first release of this package. And includes a number of features, improvements and bug fixes for pdf builds in jupyter-book projects.
Binary file ./tests/roots/test-partsToc/_build/.doctrees/environment.pickle matches
./docs/intro.md:**Improvements to LaTeX for [jupyter-book](https://jupyterbook.org/intro.html)**.
./docs/intro.md:extension to handle LaTeX builds for [jupyter-book](https://jupyterbook.org/)
./docs/intro.md:  using `myst_amsmath_enable` key of `jupyter-book`.
./.github/workflows/pdf.yml:        git clone https://github.com/executablebooks/jupyter-book.git
./.github/workflows/pdf.yml:        cd jupyter-book
./.github/workflows/pdf.yml:        path: jupyter-book/docs/_build/latex/book.pdf
./setup.cfg:    jupyter-book>=0.11,<0.12
```

Comment (mmcky) [reply setup.cfg:40]: looks like this requirement could be removed?

Comment (mmcky) [reply setup.cfg:40]: ah I see

```
ImportError while importing test module '/Users/matthewmckay/repos-collab/ebp/sphinx-jupyterbook-latex/tests/test_jupyter_book.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_jupyter_book.py:4: in <module>
    from jupyter_book.cli.main import build
E   ModuleNotFoundError: No module named 'jupyter_book'
```

Comment (mmcky) [reply setup.cfg:40]: can we update these tests to run using `sphinx_build` rather than `jupyter-book`?

Comment (mmcky) [reply setup.cfg:40]: @chrisjsewell is there any way to import these fixtures to our projects so we aren't replicating the code across our test suites. 

https://github.com/executablebooks/MyST-NB/blob/448d36f483de2bdbf2ac71d8a9b3e8fe75918be7/tests/conftest.py#L152

Maybe we need a `jupyterbook-testing` package or something like that?

Comment (mmcky) [reply setup.cfg:40]: I see there is some custom code in each of these for context though so maybe harder that it looks initially.

**Comment by mmcky** [comment]
@AakashGfude @chrisjsewell I have moved the `myst_nb` pin to only support `myst_nb >= 0.13` as I have updated the fixtures to include:

```
classes="tex2jax_ignore mathjax_ignore"
```

which comes from myst-parser update

https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#mathjax-and-math-parsing

**Comment by mmcky** [comment]
@AakashGfude what is the use of `myst` in the `options` listing in `setup.cfg`. It looks like I had to add it to testing anyway?

**Comment by mmcky** [comment]
- [ ] add `jupyter-book` test sets to `jupyter-book:sphinx4`
- [ ] Switch "building PDF of JupyterBook Docs" to use a `sphinx` build in github actions

**Comment by mmcky** [comment]
@AakashGfude @chrisjsewell are there any good `sphinx` projects (that are reasonably complex) that we can test against for the `github` action? I can make up a test repo if we need to but thought I would check.

I think removing any `jupyter-book` tests from this repo makes sense so I have:

- [x] moved the `jupyter-book` test to `jupyter-book`
- [ ] updating the `pdf.yml` workflow for github actions.

**Comment by AakashGfude** [comment]
> @AakashGfude @chrisjsewell I have moved the `myst_nb` pin to only support `myst_nb >= 0.13` as I have updated the fixtures to include:
> 
> ```
> classes="tex2jax_ignore mathjax_ignore"
> ```
> 
> which comes from myst-parser update
> 
> https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#mathjax-and-math-parsing

@mmcky the master branch of `jupyter-book` has `myst-nb~=0.12.0` pinned. https://github.com/executablebooks/jupyter-book/blob/master/setup.cfg#L44 .  But I reckon, we won't use this latest version right away anyways.

**Comment by AakashGfude** [comment]
@mmcky looks good to me.

**Review by AakashGfude — DISMISSED** [review]

**Review by AakashGfude — APPROVED** [review]

**Comment by choldgraf** [comment]
Is this waiting on anything?

**Pull request merged**